### ISSUE TITLE
validating dependent arguments

### ISSUE THREAD

**Issue by ellchow** [scallop/scallop#55]
Right now, we can validation pairs (triplets, quadruplets, etc.) but it is required that they are codependent.  I have a particular use case where I want the validation of one argument depends on the value of another argument.  Let's say I have a flag called _overwrite_ and an argument _outputPath_.  If _overwrite_ is false, validation should fail if _outputPath_ exists.  If _overwrite_ is true, validation should pass regardless of whether _outputPath_ is present (ideally, if _overwrite_ is true, I would delete the output path first and then proceed).

This does not work at the moment since _overwrite_ is required and _overwrite_ and _outputPath_ are codependent.

**Comment by Rogach** [comment]
I'm extremely sorry for responding with such delay - seems the github experienced some glitch, and I was not notified of new issue, and saw it only now.

So, the problem seems that the type of `validate` is something like `(A1, A2, A2) => Either[String, Unit]` instead of `(Option[A1], Option[A2], Option[A3]) => Either[String, Unit]`?

I can make the change to support the following syntax:

```
new ScallopConf(args) {
  val overwrite = opt[Boolean]()
  val outputPath = opt[String]()
  validateOpt(overwrite, outputPath) {
    case (Some(false), Some(out)) if new File(out).exists => Left("Output file already exists")
    case (Some(true), Some(out)) if new File(out).exists => /* delete */ Right(Unit)
    case _ => Right(Unit) // or something else
  }
}
```

Will that be better?

**Comment by Rogach** [comment]
Side note - I think that doing destructive actions (like deleting files) inside validation blocks is a bad idea. For example, if some other validation block fails (after the block about overwrite), your program wouldn't run, but the files will still be deleted - the user would be confused.

**Comment by stevedlawrence** [comment]
I'll add that I think a the validateOpt function would make some validation code that I do much simpler.

**Comment by ellchow** [comment]
yes this looks good to me.

thanks!

**Comment by Rogach** [comment]
I implemented it, and published a snapshot (for scala 2.10.x):

``` scala
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "0.9.1~5~SNAPSHOT"
```

**Comment by Rogach** [comment]
If everything is okay, I will be doing a full release today.

**Comment by stevedlawrence** [comment]
With the fix to #60, looks good to me. Thanks!

**Comment by Rogach** [comment]
You're welcome :)
I will close the issue, then, and cut the release.
